climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+
+RES_DATA = 1.0
+""" Resolution of the data if no data origin provided (km) """
+
+BLURR_STEPS = 4
+""" steps with exponential decay for fire propagation matrix """
+
+class WildFire(Hazard):
+
+    """Contains wild fires.
+
+    Wildfires comprise the challange that the definition of an event is unclear.
+    Reporting standards vary accross regions and over time. Hence, to have
+    consistency, we consider an event as a whole fire season. This allows
+    consistent risk assessment across the global and over time. Events that
+    refer to a fire season have the tag 'WFseason'.
+
+    In order to perform concrete case studies or calibrate impact functions,
+    events can be displayed as single fires. In that case they have the tag
+    'WFsingle'.
+
+
+    Attributes:
+        date_end ((np.array): integer date corresponding to the proleptic
+            Gregorian ordinal, where January 1 of year 1 has ordinal 1
+            (ordinal format of datetime library))
+
+        n_fires (np.array): number of single fires in a fire season
+
+    """
+
+    days_thres_firms = 2
+    """ Minimum number of days to consider different fires """
+
+    clus_thres_firms = 15
+    """ Clustering factor which multiplies instrument resolution """
+
+    remove_minor_fires_firms = True
+    """ removes FIRMS fires below defined theshold of entries"""
+
+    minor_fire_thres_firms = 3
+    """ number of FIRMS entries required to be considered a fire """
+
+    prop_proba = 0.21
+    """ probability of fire propagation """
+
+    max_it_propa = 500000
+    """ maximum propgation iterations for a probabilistic fire """
+
+    def __init__(self, pool=None):
+        """Empty constructor. """
+        Hazard.__init__(self, HAZ_TYPE)
+        if pool:
+            self.pool = pool
+            LOGGER.info('Using %s CPUs.', self.pool.ncpus)
+        else:
+            self.pool = None
+
+    def set_hist_fire_FIRMS(self, csv_firms, centr_res_factor=1, centroids=None):
+        """ Parse FIRMS data and generate historical fires by temporal and spatial
+        clustering.
+
+        Parameters:
+            csv_firms: csv file of the FIRMS data (https://firms.modaps.eosdis.nasa.gov/download/)
+                or pd.DataFrame of FIRMS data
+            centr_res_factor (int, optional): resolution factor with respect to
+                the satellite data to use for centroids creation. Default: 1
+            centroids (Centroids, optional): centroids in degrees to map data
+        """
+        self.clear()
+
+        # read and initialize data
+        firms = self._clean_firms_csv(csv_firms)
+        # compute centroids
+        res_data = self._firms_resolution(firms)
+        if not centroids:
+            centroids = self._firms_centroids_creation(firms, res_data, centr_res_factor)
+        else:
+            if not centroids.coord.size:
+                centroids.set_meta_to_lat_lon()
```

Comment:
This is a bad idea, as `centroids.coord` casts the `lat` `long in a 2-d numpy array for nothing. You can either check the lat or lon or both
```suggestion
            if not centroids.lat.any():
                centroids.set_meta_to_lat_lon()
```